sysfs/class_scsitape.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package sysfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+const scsitapeClassPath = "class/scsi_tape"
+
+type ScsiTapeCounters struct {
+	WriteNs      uint64 // /sys/class/scsi_tape/<Name>/stats/write_ns
+	ReadByteCnt  uint64 // /sys/class/scsi_tape/<Name>/stats/read_byte_cnt
+	IoNs         uint64 // /sys/class/scsi_tape/<Name>/stats/io_ns
+	WriteCnt     uint64 // /sys/class/scsi_tape/<Name>/stats/write_cnt
+	ResidCnt     uint64 // /sys/class/scsi_tape/<Name>/stats/resid_cnt
+	ReadNs       uint64 // /sys/class/scsi_tape/<Name>/stats/read_ns
+	InFlight     uint64 // /sys/class/scsi_tape/<Name>/stats/in_flight
+	OtherCnt     uint64 // /sys/class/scsi_tape/<Name>/stats/other_cnt
+	ReadCnt      uint64 // /sys/class/scsi_tape/<Name>/stats/read_cnt
+	WriteByteCnt uint64 // /sys/class/scsi_tape/<Name>/stats/write_byte_cnt
+}
+
+type ScsiTape struct {
+	Name     string           // /sys/class/scsi_tape/<Name>
+	Counters ScsiTapeCounters // /sys/class/scsi_tape/<Name>/statistics/*
+}
+
+type ScsiTapeClass map[string]ScsiTape
+
+// ScsiTapeClass parses st[0-9]+ devices in /sys/class/scsi_tape.
+func (fs FS) ScsiTapeClass() (ScsiTapeClass, error) {
+	path := fs.sys.Path(scsitapeClassPath)
+
+	dirs, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: unpredictable
```

Comment:
What does this comment mean?

Reply:
I replaced it with:

```
// There are n?st[0-9]+[a-b]? variants depending on device features.
// n/2 is probably overestimated but never underestimated
```

I'm new to golang but latter I found out that the "make" here might not matter much for a map. It's just for internal preallocatted structures. I'll divide by 2 for a better guess.